src/moscot/backends/ott/solver.py

```diff
@@ -340,3 +355,208 @@ def _call_kwargs(cls) -> Tuple[Set[str], Set[str]]:
+        return self._solver
+
+    @property
+    def problem_kind(self) -> ProblemKind_t:
+        """Problem kind."""
+        return "linear"
+
+    @classmethod
+    def _call_kwargs(cls) -> Tuple[Set[str], Set[str]]:
+        return {"trainloader", "validloader"}, {}  # type: ignore[return-value]
+
+
+class CondNeuralDualSolver(NeuralDualSolver):
+    """Solver class solving Conditional Neural Optimal Transport problems."""
+
+    def __init__(self, *args, cond_dim: int, **kwargs: Any) -> None:
+        super().__init__(*args, cond_dim=cond_dim, **kwargs)
+
+    def _prepare(  # type: ignore[override]
+        self,
+        distributions: DistributionCollection,
+        sample_pairs: List[Tuple[Any, Any]],
+        train_size: float = 0.9,
+        **kwargs: Any,
+    ) -> Tuple[JaxSampler, JaxSampler]:
+        train_data: List[Optional[ArrayLike]] = []
+        train_a: List[Optional[ArrayLike]] = []
+        train_b: List[Optional[ArrayLike]] = []
+        valid_data: List[Optional[ArrayLike]] = []
+        valid_a: List[Optional[ArrayLike]] = []
+        valid_b: List[Optional[ArrayLike]] = []
+
+        sample_to_idx: Dict[Any, int] = {}
+        kwargs = _filter_kwargs(JaxSampler, **kwargs)
+        if train_size == 1.0:
+            train_data = [d.xy for d in distributions]  # type:ignore[var-annotated]
+            train_a = [d.a for d in distributions]  # type:ignore[var-annotated]
+            train_b = [d.b for d in distributions]  # type:ignore[var-annotated]
+            valid_data, valid_a, valid_b = train_data, train_a, train_b
+            sample_to_idx = {k: i for i, k in enumerate(distributions)}
+        else:
+            if train_size > 1.0 or train_size <= 0.0:
+                raise ValueError("Invalid train_size. Must be: 0 < train_size <= 1")
+
+            seed = kwargs.pop("seed", 0)
+            for i, key in enumerate(distributions):
+                t_data, v_data, t_a, t_b, v_a, v_b = self._split_data(
+                    distributions[key].xy,  # type:ignore[arg-type]
+                    train_size=train_size,
+                    seed=seed,
+                    a=distributions[key].a,
+                    b=distributions[key].b,
+                )
+                train_data.append(t_data)
+                train_a.append(t_a)
+                train_b.append(t_b)
+                valid_data.append(v_data)
+                valid_a.append(v_a)
+                valid_b.append(v_b)
+                sample_to_idx[key] = i
+
+        self._train_sampler = JaxSampler(
+            train_data, sample_pairs, conditional=True, a=train_a, b=train_b, sample_to_idx=sample_to_idx, **kwargs
+        )
+        self._valid_sampler = JaxSampler(
+            valid_data, sample_pairs, conditional=True, a=valid_a, b=valid_b, sample_to_idx=sample_to_idx, **kwargs
+        )
+        return (self._train_sampler, self._valid_sampler)
+
+    def _split_data(  # type:ignore[override]
```

Comment:
This bears a lot of similarity with the parent class's `_split_data`, would unify.